src/subprocess-posix.cc

```diff
@@ -49,26 +52,32 @@ Subprocess::~Subprocess() {
 }
 
 bool Subprocess::Start(SubprocessSet* set, const string& command) {
-  int output_pipe[2];
-  if (pipe(output_pipe) < 0)
-    Fatal("pipe: %s", strerror(errno));
-  fd_ = output_pipe[0];
+  int output_pipe[2] = {-1, -1};
```

Comment:
nit: for clarity, can I ask you only define a single `subproc_stdout_fd` local variable, and only use `fd_` and `subproc_output_fd` in the rest of the code if possible, e.g.:

```
// The stdout/stderr write descriptor for non-console subprocesses.
int subproc_stdout_fd = -1;
if (use_console_) {
  fd_ = -1;
} else {
  int output_pipe[2];
  if (pipe(output_pipe) < 0) ...;
  fd_ = output_pipe[0];
  subproc_stdout_fd = output_pipe[1];
  ...
}

...

if (subproc_stdout_fd >= 0) {
  // Do not leak this descriptor to child processes. See https://github.com/ninja-build/ninja/issues/2444
  err = posix_spawn_file_actions_addclose(&action, subproc_stdout_fd);
  ...
}

....

if (subproc_stdout_fd >= 0)
  close(subproc_stdout_fd);
}
```